File: applicatie/src/main/java/com/vdzon/administratie/bankimport/regelParsers/TRTPRegelParser.java
package com.vdzon.administratie.bankimport.regelParsers;

public class TRTPRegelParser implements RegelParser {

    //TODO: deze class kan wat mooier

    @Override
    public boolean match(String omschrijving) {
        return omschrijving.startsWith("/TRTP");
    }


    @Override
    public String extractNaam(String omschrijving) {
        String[] split = omschrijving.split("/");
        for (int i = 0; i < split.length; i++) {
            if (split[i].equals("NAME")) {
                return split[i + 1];
            }
        }
        return "Onbekend";
    }

    @Override
    public String extractOmschrijving(String omschrijving) {
        String[] split = omschrijving.split("/");
        for (int i = 0; i < split.length; i++) {
            if (split[i].equals("REMI")) {
                return split[i + 1];
            }
        }
        return omschrijving;
    }
}

File: bankimport/impl/src/main/java/com/vdzon/administratie/abnamrobankimport/regelParsers/TRTPRegelParser.kt
package com.vdzon.administratie.abnamrobankimport.regelParsers

class TRTPRegelParser : RegelParser {

    //TODO: deze class kan wat mooier

    override fun match(omschrijving: String): Boolean = omschrijving.startsWith("/TRTP")

    override fun extractNaam(omschrijving: String): String {
        val split = omschrijving.split("/".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        for (i in split.indices) {
            if (split[i] == "NAME") {
                return split[i + 1]
            }
        }
        return "Onbekend"
    }

    override fun extractOmschrijving(omschrijving: String): String {
        val split = omschrijving.split("/".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        for (i in split.indices) {
            if (split[i] == "REMI") {
                return split[i + 1]
            }
        }
        return omschrijving
    }
}
